core/services/ardupilot_manager/main.py

```diff
@@ -1,6 +1,34 @@
 #! /usr/bin/env python3
+from typing import Any, Dict, List
+
+import uvicorn
 from ArduPilotManager import ArduPilotManager
+from fastapi import FastAPI
+from mavlink_proxy.Endpoint import Endpoint, EndpointType
+
+app = FastAPI()
+autopilot = ArduPilotManager()
+
+
+@app.get("/available_endpoints", response_model=List[Dict[str, Any]])
+def get_available_endpoints() -> Any:
+    return [endpoint.__dict__ for endpoint in autopilot.get_endpoints()]
+
+
+@app.post("/endpoint", response_model=bool)
+def create_new_endpoint(connection_type: EndpointType, place: str, argument: str = "") -> Any:
```

Comment:
It's not validating place and argument:
```
[{"connType":"udpin","place":"0.0.0.0","argument":"14550"},{"connType":"udpin","place":"potato","argument":"-1234"},{"connType":"undefined","place":"5","argument":"what"}]
```

Reply:
I think the validation is being done inside the Endpoint class, during instantiation. It is just not that good yet. We should improve it on the class itself, instead of passing the responsibility to the REST API, right?

In this case, I would suggest using a [BaseModel](https://pydantic-docs.helpmanual.io/usage/models/) instead of a dataclass for Endpoint, as BaseModel is much more powerful, especially with validations.

Reply:
Edit: I think we can use a [pydantic dataclass](https://pydantic-docs.helpmanual.io/usage/dataclasses/) instead of BaseModel. It's a drop-in replacement for the standard dataclass, but comes with automatic parsing from FastAPI and enhanced validation.